src/plugins/teleop/Teleop_TEST.cc

```diff
@@ -0,0 +1,442 @@
+/*
+ * Copyright (C) 2017 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <gtest/gtest.h>
+#ifdef _MSC_VER
+#pragma warning(push, 0)
+#endif
+#include <ignition/msgs/stringmsg.pb.h>
+#ifdef _MSC_VER
+#pragma warning(pop)
+#endif
+
+#include <ignition/utilities/ExtraTestMacros.hh>
+
+#include "test_config.h"  // NOLINT(build/include)
+#include "ignition/gui/Application.hh"
+#include "ignition/gui/Plugin.hh"
+#include "ignition/gui/MainWindow.hh"
+#include <ignition/gui/qt.h>
+
+#include "Teleop.hh"
+
+int g_argc = 1;
+char **g_argv = new char *[g_argc];
+
+using namespace ignition;
+using namespace gui;
+
+class TeleopTest : public ::testing::Test {
+
+ protected:
+  void SetUp() override {
+    common::Console::SetVerbosity(4);
+  }
+
+  /// \param[in] _topic Topic to publish twist messages.
+  protected: void TestButtons(const std::string &_topic)
+  {
+    Application app(g_argc, g_argv);
+    app.AddPluginPath(std::string(PROJECT_BINARY_PATH) + "/lib");
+
+    // Load plugin
+    const char *pluginStr =
+      "<plugin filename=\"Teleop\">"
+        "<ignition-gui>"
+          "<title>Teleop!</title>"
+        "</ignition-gui>"
+      "</plugin>";
+
+    tinyxml2::XMLDocument pluginDoc;
+    EXPECT_EQ(tinyxml2::XML_SUCCESS, pluginDoc.Parse(pluginStr));
+    EXPECT_TRUE(app.LoadPlugin("Teleop",
+        pluginDoc.FirstChildElement("plugin")));
+
+    // Get main window
+    auto win = app.findChild<MainWindow *>();
+    ASSERT_NE(nullptr, win);
+
+    // Show, but don't exec, so we don't block
+    win->QuickWindow()->show();
+
+    // Get plugin
+    auto plugins = win->findChildren<plugins::Teleop *>();
```

Comment:
Can this be const?

Reply:
That variable takes a value which can't be constant, since I clear it out at the end of the test.